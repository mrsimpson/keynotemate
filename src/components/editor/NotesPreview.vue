<template>
  <div class="notes-preview">
    <h2>Preview</h2>
    <div class="notes-grid">
      <div v-for="(note, index) in notes" :key="index" class="note-item">
        <div class="note-pitch">{{ note.pitch }}{{ note.octave }}</div>
        <div class="note-duration">Duration: {{ note.duration }}</div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import type {ParsedNote} from '../../types/musicxml';

defineProps<{
  notes: ParsedNote[];
}>();
</script>

<style scoped>
.notes-preview {
  margin-top: 2rem;
  padding-top: 2rem;
  border-top: 1px solid #eee;
}

.notes-preview h2 {
  margin-bottom: 1rem;
  color: #333;
}

.notes-grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(120px, 1fr));
  gap: 1rem;
}

.note-item {
  background: #f8f9fa;
  padding: 1rem;
  border-radius: 6px;
  text-align: center;
}

.note-pitch {
  font-size: 1.25rem;
  font-weight: bold;
  color: #42b883;
}

.note-duration {
  margin-top: 0.5rem;
  font-size: 0.875rem;
  color: #666;
}
</style>
