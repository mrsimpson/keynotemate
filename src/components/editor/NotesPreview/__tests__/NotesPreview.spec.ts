import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import type { ParsedNote } from '@/types/musicxml.ts'
import NotesPreview from '@/components/editor/NotesPreview.vue'

describe('NotesPreview Component', () => {
  const mockNotes: ParsedNote[] = [
    {
      pitch: 'C',
      duration: {
        divisions: 4,
        type: 'quarter',
        relativeLength: 1,
      },
      octave: 4,
    },
    {
      pitch: 'D',
      duration: {
        divisions: 4,
        type: 'quarter',
        relativeLength: 1,
      },
      octave: 4,
    },
  ]

  it('renders notes preview snapshot', () => {
    const wrapper = mount(NotesPreview, {
      props: {
        notes: mockNotes,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders empty notes preview snapshot', () => {
    const wrapper = mount(NotesPreview, {
      props: {
        notes: [],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
