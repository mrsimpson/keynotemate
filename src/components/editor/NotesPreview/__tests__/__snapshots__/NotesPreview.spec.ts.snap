// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotesPreview Component > renders empty notes preview snapshot 1`] = `
"<div data-v-85ae9d83="" class="notes-preview">
  <h2 data-v-85ae9d83="">Preview</h2>
  <div data-v-85ae9d83="" class="notes-grid"></div>
</div>"
`;

exports[`NotesPreview Component > renders notes preview snapshot 1`] = `
"<div data-v-85ae9d83="" class="notes-preview">
  <h2 data-v-85ae9d83="">Preview</h2>
  <div data-v-85ae9d83="" class="notes-grid">
    <div data-v-85ae9d83="" class="note-item">
      <div data-v-85ae9d83="" class="note-pitch">C4</div>
      <div data-v-85ae9d83="" class="note-duration">Duration: {
        "divisions": 4,
        "type": "quarter",
        "relativeLength": 1
        }</div>
    </div>
    <div data-v-85ae9d83="" class="note-item">
      <div data-v-85ae9d83="" class="note-pitch">D4</div>
      <div data-v-85ae9d83="" class="note-duration">Duration: {
        "divisions": 4,
        "type": "quarter",
        "relativeLength": 1
        }</div>
    </div>
  </div>
</div>"
`;
